### ISSUE TITLE
Post Fails Using HttpClient in RestSharp Client constructor

### ISSUE THREAD

**Issue by silenuz** [restsharp/RestSharp#1837]
While working with an API recently I encountered unexpected behavior when using an existing HttpClient in the RestClient constructor.

If I use the HttpClient in the constructor all my posts return with a bad request response (code 400), however creating the RestClient with the same base url as used in the HttpClient works.

**To Reproduce**

TestClass:
```
public class Connection {
    RestClient client;

    public Connection(bool wantFail)
    {
        string     baseUrl = @"https://travellermap.com/api/";
        HttpClient http    = new HttpClient();
        http.BaseAddress = new Uri(baseUrl);

        if (wantFail) {
            client = new RestClient(http);
        }
        else {
            client = new RestClient(baseUrl);
        }
    }

    public byte[] CreateSectorMapTest() {
        byte[] result;

       
        var request = new RestRequest("poster?subsector=A", Method.Post);
        request.AddHeader("Content-Type", "text/plain");
        request.AddHeader("Accept", "image/svg+xml");

        var body = @"#--------1---------2---------3---------4---------5-------
#PlanetName   Loc. UPP Code   B   Notes         Z  PBG Al
#----------   ---- ---------  - --------------- -  --- --
Cogrikonur    0101 C8C5446-9    Fl Ni              120 Im";

        request.AddParameter("text/plain", body, ParameterType.RequestBody);
        RestResponse response = client.Execute(request);

        if (response.IsSuccessful) {
            result = response.RawBytes;
        }
        else {
            result = null;
        }

        return result;
    }
}
```

Test Application Console:

```
Console.WriteLine("Hello, World!");

Console.WriteLine("Testing For Success");

Connection goodConnection = new Connection(false);
byte[] image = goodConnection.CreateSectorMapTest();

if (image != null) {
    Console.WriteLine("Good Connection Worked");
}
else {
    Console.WriteLine("Good Connection Failed");
}

Console.WriteLine("Testing For Fail");
Connection badConnection = new Connection(true);
byte[] imageBad = badConnection.CreateSectorMapTest();

if (imageBad != null) {
    Console.WriteLine("Bad Connection Worked");
}
else {
    Console.WriteLine("Bad Connection Failed");
}

Console.ReadLine();

Console.WriteLine("Finished");

```
Output of Console App:
Hello, World!
Testing For Success
Good Connection Worked
Testing For Fail
Bad Connection Fail

**Expected behavior**

I would expect that the two constructors should provide the same functionality.

**Desktop (please complete the following information):**
 - OS: [e.g. Linux]
 - .NET version [e.g. .NET 6]
 - Version [e.g. 107.0.3.1-alpha.0.24] + in Clone of current repo

**Comment by alexeyzimarev** [comment]
You can use requestbin.com to trace the request in both cases, maybe you can spot the difference.

And, again, do not add the content type header, don't use `request.AddParameter("text/plain", body, ParameterType.RequestBody);`. Use `request.AddStringBody(body, "text/plain")` instead.

**Comment by silenuz** [comment]
Well I changed the code to this: 

```
public byte[] CreateSectorMapTest() {
        byte[] result;

       
        var request = new RestRequest("poster?subsector=A", Method.Post);
        request.AddHeader("Accept", "image/svg+xml");

        var body = @"#--------1---------2---------3---------4---------5-------
#PlanetName   Loc. UPP Code   B   Notes         Z  PBG Al
#----------   ---- ---------  - --------------- -  --- --
Cogrikonur    0101 C8C5446-9    Fl Ni              120 Im";

        request.AddStringBody(body, "text/plain");
        RestResponse response = client.Execute(request);

        if (response.IsSuccessful) {
            result = response.RawBytes;
        }
        else {
            result = null;
        }

        return result;
    }
```
As you can see I removed the content type from the header and changed the code to add the body.  This however does not affect the outcome at all, if the RestClient is instantiated with the HttpClient it fails, with the same Bad Request:

```
StatusCode: BadRequest, Content-Type: text/plain, Content-Length: 55
System.Net.Http.HttpRequestException: Request failed with status code BadRequest
```

**Comment by alexeyzimarev** [comment]
My suggestions were about using the API. As for your issue, if you read my reply from the start, I suggested tracing the request with those versions you have and trying to spot the difference.

**Comment by silenuz** [comment]
Your suggestions were some of the first things I tried, but there are no obvious differences, or even subtle ones that I can find.  But it seems obvious that if it an http client can be used in the constructor, then RestSharp should work with the HttpClient.

As I provided complete code that demonstrates that given the exact same message creation process that the RestClient fails if it was created with an http client, but succeeds if it wasn't. Exact same methods, exact same client, exact same address, only difference is creating the RestClient with an HttpClient or creating it with a base URL.

Also note that creating the RestClient with an HttpClient that has no base address set will result in an exception.  On a related note so far that is the only difference found in the two instances, when RestClient creates it's own HttpClient, at the point the request is made (ExecuteInternal) the HttpClient created by the RestClient has an empty base address, whereas the HttpClient passed in the constructor has whatever base address was set.

**Comment by alexeyzimarev** [comment]
> passing an HttpClient to the RestClient constructor that does not have a base address set will cause a fatal exception

Can you elaborate on that, please? It should not fail.

**Comment by silenuz** [comment]
If a HttpClient is passed to the constructor and it does not have abase address set it results in:

```
System.ArgumentOutOfRangeException: Request resource doesn't contain a valid scheme for an empty base URL of the client (Parameter 'request')
   at RestSharp.RestClient.DoBuildUriValidations(RestRequest request) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/RestClient.cs:line 218
   at RestSharp.RestClient.BuildUri(RestRequest request) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/RestClient.cs:line 192
   at RestSharp.RestClient.ExecuteInternal(RestRequest request, CancellationToken cancellationToken) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/RestClient.Async.cs:line 57
   at RestSharp.RestClient.ExecuteAsync(RestRequest request, CancellationToken cancellationToken) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/RestClient.Async.cs:line 26
   at RestSharp.AsyncHelpers.<>c__DisplayClass1_0`1.<<RunSync>b__0>d.MoveNext() in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/AsyncHelpers.cs:line 47
--- End of stack trace from previous location ---
   at RestSharp.AsyncHelpers.CustomSynchronizationContext.<Run>g__PostCallback|7_0(Object _) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/AsyncHelpers.cs:line 84
   at RestSharp.AsyncHelpers.CustomSynchronizationContext.Run() in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/AsyncHelpers.cs:line 103
   at RestSharp.AsyncHelpers.RunSync(Func`1 task) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/AsyncHelpers.cs:line 32
   at RestSharp.AsyncHelpers.RunSync[T](Func`1 task) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/AsyncHelpers.cs:line 47
   at RestSharp.RestClient.Execute(RestRequest request) in /home/jordan/Code/DotNet/RestSharp/src/RestSharp/Sync/RestClient.Sync.cs:line 23
   at RestTest.Connection.CreateSectorMapTest() in /home/jordan/Code/DotNet/RestSharp/RestTest/Connection.cs:line 51
```

Also note as per my original message that the Base URL still has to be explicitly set even if the HttpClient bas eaddress is set or some posts cause a bad request issue.

**Comment by alexeyzimarev** [comment]
Ok, then it's a duplicate of https://github.com/restsharp/RestSharp/issues/1765. It was fixed, but not released as a stable version. If you use the latest preview, you will get it working.

**Comment by silenuz** [comment]
I just cloned the Repo, and I still get the exact same behavior as before.   So to me this is not fixed.

**Comment by erlendthune** [comment]
I had the same problem as described in the original question. In my case it turned out to be because the header "User-Agent" is not set by default in HttpClient whereas it turns out it is set by default by RestSharp. Setting this header on the HttpClient before passing it to the RestSharp constructor solved the problem for me, and it also made HttpClient work on its own. 

The reason I started using RestSharp was because HttpClient failed. I then used RestSharp with a base address, but couldn't sleep before I had figured out the reason why HttpClient failed. In my specific case I was trying to download an ics calendar file from https://outlook.office365.com/ The code I use is here: https://github.com/erlendthune/RememberCalendar
